dpnp/linalg/dpnp_utils_linalg.py

```diff
@@ -164,3 +207,151 @@ def dpnp_eigh(a, UPLO):
+            else:
+                b = b.reshape(prod(b_shape[:-1]), b_shape[-1])
+
+            a = a.reshape(prod(a_shape[:-2]), a_shape[-2], a_shape[-1])
+
+            a_usm_arr = dpnp.get_usm_ndarray(a)
+            b_usm_arr = dpnp.get_usm_ndarray(b)
+            reshape = True
+
+        batch_size = a.shape[0]
+
+        coeff_vecs = [None] * batch_size
+        val_vecs = [None] * batch_size
+        a_ht_copy_ev = [None] * batch_size
+        b_ht_copy_ev = [None] * batch_size
+        ht_lapack_ev = [None] * batch_size
+
+        for i in range(batch_size):
+            # oneMKL LAPACK assumes fortran-like array as input, so
+            # allocate a memory with 'F' order for dpnp array of coefficient matrix
+            # and multiple dependent variables array
+            coeff_vecs[i] = dpnp.empty_like(
+                a[i], order="F", dtype=res_type, usm_type=res_usm_type
+            )
+            val_vecs[i] = dpnp.empty_like(
+                b[i], order="F", dtype=res_type, usm_type=res_usm_type
+            )
+
+            # use DPCTL tensor function to fill the coefficient matrix array
+            # and the array of multiple dependent variables with content
+            # from the input arrays
+            a_ht_copy_ev[i], a_copy_ev = ti._copy_usm_ndarray_into_usm_ndarray(
+                src=a_usm_arr[i],
+                dst=coeff_vecs[i].get_array(),
+                sycl_queue=a.sycl_queue,
+            )
+            b_ht_copy_ev[i], b_copy_ev = ti._copy_usm_ndarray_into_usm_ndarray(
+                src=b_usm_arr[i],
+                dst=val_vecs[i].get_array(),
+                sycl_queue=b.sycl_queue,
+            )
+
+            # Call the LAPACK extension function _gesv to solve the system of linear
+            # equations using a portion of the coefficient square matrix and a
+            # corresponding portion of the dependent variables array.
+            ht_lapack_ev[i], _ = li._gesv(
+                exec_q,
+                coeff_vecs[i].get_array(),
+                val_vecs[i].get_array(),
+                depends=[a_copy_ev, b_copy_ev],
+            )
+
+        for i in range(batch_size):
+            ht_lapack_ev[i].wait()
+            b_ht_copy_ev[i].wait()
+            a_ht_copy_ev[i].wait()
+
+        # combine the list of solutions into a single array
+        out_v = dpnp.array(val_vecs, order=b_order, usm_type=res_usm_type)
+        if reshape:
+            # shape of the out_v must be equal to the shape of the array of
+            # dependent variables
+            out_v = out_v.reshape(orig_shape_b)
+        return out_v
+    else:
+        # oneMKL LAPACK assumes fortran-like array as input, so
+        # allocate a memory with 'F' order for dpnp array of coefficient matrix
+        # and multiple dependent variables
+        a_f = dpnp.empty_like(
+            a, order="F", dtype=res_type, usm_type=res_usm_type
+        )
+        b_f = dpnp.empty_like(
```

Comment:
Will it make sense from performance view if you move copying just after the memory allocation? like:
```python
a_f = dpnp.empty_like(...)
# copy a to a_f

if need_copy_for_b:
    b_f = dpnp.empty_like(...)
    # copy a to b_f
```
In that case I believe the allocation memory for `b` array might be done in parallel with copying of input array `a` into `a_f` temporary.